app/scripts/metamask-controller.js

```diff
@@ -114,6 +115,19 @@ export default class MetamaskController extends EventEmitter {
       migrateAddressBookState: this.migrateAddressBookState.bind(this),
     })
 
+    // This depends on preferences controller state
+    this.trackSegmentEvent = getTrackSegmentEvent(
+      this.platform.getVersion(),
+      () => this.preferencesController.getParticipateInMetaMetrics(),
+      () => {
+        const {
+          currentLocale = '',
+          metaMetricsId = '',
```

Comment:
Nit: I don't think these defaults are required; `currentLocale` should always be set, and the default value of `metaMetricsId` of `null` seems more appropriate than `''` as a default anyway in the off chance we have a bug where that somehow isn't set.

```suggestion
          currentLocale,
          metaMetricsId,
```